adapters/oidc/spring-boot-adapter-core/src/main/java/org/keycloak/adapters/springboot/KeycloakBaseSpringBootConfiguration.java

```diff
@@ -196,6 +185,21 @@ public void customize(Server server) {
 
             webAppContext.setSecurityHandler(securityHandler);
         }
+
+        private WebAppContext getWebAppContext(Handler... handlers) {
+            for (Handler handler : handlers) {
+                if (handler instanceof WebAppContext) {
+                    return (WebAppContext) handler;
+                } else if (handler instanceof HandlerList) {
```

Comment:
`HandlerList` is a `HandlerCollection` so you could remove that if statement I believe. 

Have you considered using reflection as well instead of using `instanceof` ? Looks like HandlerList, HandlerCollection and HandlerWrapper all have the same method and signature